### PR TITLE
Add NuGet package READMEs

### LINKED ISSUE #6991: Add readmes to NuGet packages
NuGet.org now [prominently features READMEs][1]. The NuGet packages MSBuild produces should use this to highlight a few things:

1. Links to the docs.
2. Which packages are the right ones to reference for task authors.
2. Which packages are more or less "internal implementation details" (`Runtime`) and what you should use instead (MSBuildLocator).

After we get a draft out we should see if we can get some of @ghogen's time for editing.

[1]: https://devblogs.microsoft.com/nuget/introducing-the-new-nuget-org-package-details-page/

### PR DESCRIPTION
**Pull request by rainersigwald**
This should improve understandablity of our packages.

Fixes #6991.

Details on the format:

* https://docs.microsoft.com/nuget/nuget-org/package-readme-on-nuget-org
* https://devblogs.microsoft.com/nuget/add-a-readme-to-your-nuget-package/

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -1,27 +1,29 @@
 <Project>
 
   <PropertyGroup>
+    <PackageReadmeFile>README.md</PackageReadmeFile>
```
Comment (baronfel) [inline Directory.Build.targets:4]: I love this trick, I do it on all of my multi-package projects :D

**Review by baronfel — APPROVED** [review]
LGTM, thanks for doing this :) I built locally with `./build.cmd -pack` and verified the packages via nuget.info, so this is all set from my perspective.

**Comment by baronfel** [comment]
After this PR, should [this doc](https://github.com/dotnet/msbuild/blob/main/documentation/consuming-nuget-package.md) still exist?

**Comment by rainersigwald** [comment]
> After this PR, should [this doc](https://github.com/dotnet/msbuild/blob/main/documentation/consuming-nuget-package.md) still exist?

Hm, I think having the pointers to the packages still makes sense but it can be streamlined. I'm also going to steal some of that text for the new READMEs 🤦🏻

File: documentation/consuming-nuget-package.md [hunk 2]
```diff
@@ -11,42 +11,42 @@ The MSBuild team currently publishes five NuGet packages.  Our packages are publ
...
-appropriate parameters for your project.  To retrieve evaluated items, call methods such as  properties such as [GetItem](https://msdn.microsoft.com/en-us/library/microsoft.build.evaluation.project.getitems.aspx)
-or [GetPropertyValue](https://msdn.microsoft.com/en-us/library/microsoft.build.evaluation.project.getpropertyvalue.aspx).
+Developers should reference this package to create, edit, evaluate, or build MSBuild projects.
 
 ## Microsoft.Build.Runtime
-This package contains the standard set of MSBuild projects which are imported by other projects such as CSharp and Visual Basic as well as the MSBuild executable.  Developers should reference this package if they want to
-redistribute the MSBuild runtime to evaluate or build MSBuild projects within their application.  This can be necessary because prior to MSBuild version 15, MSBuild was installed globally on a machine and universally
-available to all applications.  However, in MSBuild version 15 and forward, MSBuild is redistributed by each application that uses it and applications are unable to share other instances.  
+
+This package delivers a complete executable copy of MSBuild. Reference this
```
Comment (Forgind) [inline documentation/consuming-nuget-package.md:43]: I don't think I'd ever heard of this 😮

**Review by Forgind — APPROVED** [review]

**Pull request merged**